### PR TITLE
Multiple networks: Switch safe adjustments

### PR DESCRIPTION
**Pull request by elgatovital**
Handles #1536

Changes proposed in this pull request:
- Load chain data for safe 
- Group safes by chain
- Apply chain theming 

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/res/drawable/circle.xml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <shape xmlns:android="http://schemas.android.com/apk/res/android"
     android:shape="oval">
-    <solid android:color="@color/logo_background" />
+    <solid android:color="@color/primary" />
```
Comment (biafra23) [inline app/src/main/res/drawable/circle.xml:4]: This looks wrong.

File: app/src/main/res/layout/dialog_share_safe.xml [hunk 2]
```diff
@@ -22,6 +22,43 @@
...
+                android:layout_width="24dp"
+                android:layout_height="24dp"
+                android:background="@drawable/background_selectable"
+                android:src="@drawable/ic_link_green_24dp"
+                app:layout_constraintBottom_toBottomOf="@id/safe_address"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toEndOf="@id/safe_address"
+                app:layout_constraintTop_toTopOf="@id/safe_address" />
+
+            <ImageView
```
Comment (biafra23) [inline app/src/main/res/layout/dialog_share_safe.xml:49]: The outline looks incomplete:

image

File: app/src/main/res/layout/item_chain_header.xml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        android:layout_marginEnd="8dp"
+        android:layout_marginStart="8dp"
+        android:layout_marginTop="8dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <aapt:attr name="android:background">
+
```
Comment (biafra23) [inline app/src/main/res/layout/item_chain_header.xml:25]: Several unnecessary newlines?

File: app/src/main/res/layout/item_safe.xml [hunk 4]
```diff
@@ -26,7 +26,7 @@
         android:layout_width="0dp"
         android:layout_height="wrap_content"
         android:layout_marginHorizontal="@dimen/default_margin"
-        android:layout_marginTop="@dimen/default_margin"
+        android:layout_marginTop="8dp"
```
Comment (biafra23) [inline app/src/main/res/layout/item_safe.xml:29]: Why not `@dimen/default_small_margin` here and below?

File: app/src/main/java/io/gnosis/safe/ui/splash/SplashViewModel.kt [hunk 5]
```diff
@@ -40,6 +42,14 @@ constructor(
         tracker.setNumSafes(numSafes)
         tracker.setNumKeysImported(credentialsRepository.ownerCount(Owner.Type.IMPORTED))
         tracker.setNumKeysGenerated(credentialsRepository.ownerCount(Owner.Type.GENERATED))
+
+        kotlin.runCatching {
+            val chainInfos = chainInfoRepository.getChainInfo()
```
Comment (biafra23) [inline app/src/main/java/io/gnosis/safe/ui/splash/SplashViewModel.kt:47]: Why move this to splash?

File: app/src/main/res/layout/item_chain_header.xml (hunk 3 above)
Comment (elgatovital) [reply app/src/main/res/layout/item_chain_header.xml:25]: just one. One empty line between parent and child tag is acceptable and makes xml more legible

File: app/src/main/java/io/gnosis/safe/ui/splash/SplashViewModel.kt (hunk 5 above)
Comment (elgatovital) [reply app/src/main/java/io/gnosis/safe/ui/splash/SplashViewModel.kt:47]: we have to load chain details only on app start and not every time app comes back from background. Also having this in view model makes it easier to verify and test

File: app/src/main/java/io/gnosis/safe/ui/StartActivity.kt [hunk 6]
```diff
@@ -252,8 +235,14 @@ class StartActivity : BaseActivity(), SafeOverviewNavigationHandler, AppStateLis
...
             chainRibbon.visible(true)
-            //TODO: get chain name and ribbon color from safe data
-            chainRibbon.text = BuildConfig.BLOCKCHAIN_NAME
+            safe.chain?.let {
+                chainRibbon.text = it.name
+                chainRibbon.setTextColor(Color.parseColor(it.textColor))
+                chainRibbon.setBackgroundColor(Color.parseColor(it.backgroundColor))
+            } ?: run {
+                chainRibbon.text = ""
+                chainRibbon.setBackgroundColor(getColorCompat(R.color.primary))
```
Comment (biafra23) [inline app/src/main/java/io/gnosis/safe/ui/StartActivity.kt:244]: So primary (green) is the default (and no chain name) if we don't have any chain info?

**Review by biafra23 — APPROVED** [review]

File: app/src/main/java/io/gnosis/safe/ui/StartActivity.kt (hunk 6 above)
Comment (elgatovital) [reply app/src/main/java/io/gnosis/safe/ui/StartActivity.kt:244]: At least we will have default values for mainnet that will be added in the next pr. Edge case when we don't have chain info for a safe should not be possible

**Pull request merged**